Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,10 +761,18 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+
+        # if pstate can't be found
+        if not pstate.get('state', 'engineresumedata'):
```

Comment:
All who called `save_download_pstate` also called `network_checkpoint` beforehand. `network_checkpoint` empties `engineresumedata`, so whenever we called `save_download_pstate` (in this stage), it will wait an alert before actually write it to file.